core/state/state_object.go

```diff
@@ -410,15 +377,10 @@ func (s *stateObject) updateTrie() (Trie, error) {
 		}
 		s.db.StorageDeleted.Add(1)
 	}
-	// If no slots were touched, issue a warning as we shouldn't have done all
-	// the above work in the first place
-	if len(usedStorage) == 0 {
-		log.Error("State object update was noop", "addr", s.address, "slots", len(s.pendingStorage))
-	}
 	if s.db.prefetcher != nil {
-		s.db.prefetcher.used(s.addrHash, s.data.Root, usedStorage)
+		s.db.prefetcher.used(s.addrHash, s.data.Root, used)
 	}
-	s.pendingStorage = make(Storage) // reset pending map
+	s.needCommit = make(Storage) // empties the commit markers
```

Comment:
Might want to use some map clearing method here to retain the memory and not reallocate for the next tx (pre byzantium).

Reply:
maps.Delete is still in `golang/exp` package. Probably wait a bit until it's fully shipped in standard `maps` package?

Reply:
This one?

https://pkg.go.dev/builtin#clear